### DISCUSSION TITLE
Could we see this data fed into HA via some sort of integration?

### DISCUSSION THREAD

**Discussion by daernsinstantfortress** [mmcc-xx/WhosAtMyFeeder#10] [Ideas]
Mostly just wanted to say thanks for this excellent project. My wife and I have been playing around testing it today and while we've mostly only caught magpies, it's been able to show us that this is a really cool thing with loads of potential. Next project for us is to improve the aesthetics of our "smart bird feeder" ;-)
![image](https://github.com/mmcc-xx/WhosAtMyFeeder/assets/7900720/27d809d1-8421-4f90-a388-5d71503daa1f)

While I'm trying to work out how on earth I can improve such a magnificent setup, here's one question and one suggestion:

Question: Do you have any recommendations for cameras that work well with bird feeders, or can be easily integrated? The one we've been messing with today (a rubbish, cheap one I usually use for watching the progress of 3d prints!) has shown us that something with higher resolution would be well worth using and we're pondering whether a feeder-integrated one would be better, or a longer range one mounted facing the feeder. Was thinking [something like this](https://www.amazon.co.uk/Reolink-Security-Detection-Spotlight-Time-Lapse/dp/B0BL3G4TSP/) for a remote camera as it's got a decent zoom on it and is PoE so relatively easy to install. Interesting in your thoughts.

Suggestion: It would be great to see this data integrated into Home Assistant in some way. An MQTT stream of recognition events would be great, but a full UI integration would be even better!

**Comment by mmcc-xx** [comment]
I love your setup. Mine is similar, though I am using an outdoor rated camera :) I made a stand with some lumber from the home improvement store. My wife rolls her eyes a bit. 

I've been focusing lately on my companion app that detects birds by sound, and it is much further along. It has MQTT support, and reports, and lots of other stuff. If you camera provides audio, you can do both.

https://github.com/mmcc-xx/BirdCAGE

I intend to bring this app up to the same level.

I don't know a whole lot about the cameras out there. I got one of these...
https://www.amazon.com/gp/product/B08QV8T3D9/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

I picked it because it has a zoom lens and I wanted something I could get a narrower field of view with so more pixels would be used on the birds.

If you go with this camera, know that the high res rstp stream isn't great. I'm using its high res rtmp stream to feed Frigate and it is much more stable. I use the low res rstp stream to feed my audio app, as it has the same audio as the high res stream, which is to say "not great but adequate".

**Reply by daernsinstantfortress** [reply]
Ah, suspiciously that was the exact camera I was looking at myself, albeit the PoE version. And for exactly the same reason - 5x optical zoom sounds a good thing for getting tight on the feeders.

What sort of distance do you have it mounted? I was struggling to get decent information on things like minimum focal distance and the suchlike, so it's a bit of a guess.

Thanks for the nudge on the bird sound detection app - will give that a try too!
